### PR TITLE
filter resources by tags

### PR DESCRIPTION
**Pull request by riteshsonawane1372**
## Fixes Issue

#18

### CODE REVIEW

**Comment by riteshsonawane1372** [comment]
``awsctl get ec2 -t key``
``awsctl get ec2 -t key=value1,value2,value3....``

File: cmd/s3.go [hunk 1]
```diff
@@ -26,38 +24,23 @@ var s3Cmd = &cobra.Command{
...
-	}
 
-	// Create an Amazon S3 service client
-	client := s3.NewFromConfig(cfg)
+	w := tabwriter.NewWriter(os.Stdout, 10, 5, 2, ' ', tabwriter.TabIndent)
+	defer w.Flush()
+	fmt.Fprintln(w, "NAME", "\t", "CREATED_AT", "\t", "SIZE")
 
-	out, err := client.ListBuckets(ctx, &s3.ListBucketsInput{})
+	bucketList, err := s3API.GetBucket()
```
Comment (surajnarwade) [inline cmd/s3.go:32]: keep s3 changes in different PR

**Comment by surajnarwade** [comment]
@riteshsonawane1372 only remove the changes you've added, with new commit, you're removing entire s3 code which is functional

File: pkg/EC2/getEc2Info.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+				}
+			}
+		}
+	}
+
+	return ec2Instance
+
+}
+
+func DescribeInstance() (*ec2.DescribeInstancesOutput, error) {
```
Comment (surajnarwade) [inline pkg/EC2/getEc2Info.go:102]: can we return `EC2Instance` instead of whole output?

Comment (riteshsonawane1372) [reply pkg/EC2/getEc2Info.go:102]: Okay

File: pkg/EC2/getEc2Info.go [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+	InstanceType      string
+	InstancePrivateIP string
+}
+
+func GetEC2Instance() []EC2Instance {
+
+	var ec2Instance []EC2Instance
+	info, err := DescribeInstance()
+	if err != nil {
+		log.Fatal("Error Occured while retrieving Information", err)
```
Comment (surajnarwade) [inline pkg/EC2/getEc2Info.go:24]: `Error occured while retrieving information`

File: pkg/EC2/getEc2Info.go (hunk 2 above)
Comment (riteshsonawane1372) [reply pkg/EC2/getEc2Info.go:102]: @surajnarwade

File: pkg/EC2/getEc2Info.go (hunk 3 above)
Comment (surajnarwade) [reply pkg/EC2/getEc2Info.go:24]: please update the error message as shown

File: pkg/EC2/getEc2Info.go (hunk 2 above)
Comment (surajnarwade) [reply pkg/EC2/getEc2Info.go:102]: in any of these functions you are only dealing with values are part of struct `EC2Instance` so why not add tags field to it as well

File: cmd/ec2.go [hunk 4]
```diff
@@ -6,37 +6,75 @@ package cmd
...
 var ec2Cmd = &cobra.Command{
 	Use:   "ec2",
 	Short: "Print ec2 related information",
 	Long: `For example,
 		awsctl get ec2`,
-	Run: getEC2command,
+	Run: func(cmd *cobra.Command, args []string) {
+		tag, _ := cmd.Flags().GetString("tags")
+		if tag != "" {
+			getEc2instanceTag(cmd, args)
```
Comment (surajnarwade) [inline cmd/ec2.go:24]: since both the functions are pretty much the same with minor feature change such as tag, can we use only one function and also pass `flags` to the same function.

something like getEC2(cmd, args,flags)

Comment (surajnarwade) [reply cmd/ec2.go:24]: ignore in fact, you can fetch flags from `cmd` itself so no need of checking tags at this stage

File: pkg/ec2/getEc2Info.go [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+						})
+					}
+				}
+			}
+		}
+	}
+	return ec2Instance
+}
+
+func Ec2Client() (*ec2.Client, context.Context) {
```
Comment (surajnarwade) [inline pkg/ec2/getEc2Info.go:61]: just for the better reading purposes, usually we put context first

File: pkg/ec2/getEc2Info.go [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+						})
+					}
+				}
+			}
+		}
+	}
+	return ec2Instance
+}
+
+func Ec2Client() (context.Context,*ec2.Client, ) {
```
Comment (surajnarwade) [inline pkg/ec2/getEc2Info.go:61]: extra comma
